main.c

```diff
@@ -392,12 +392,18 @@ void update_info(void)
 	}
 }
 
-int ptr_third_x(void)
+int nav_button(void)
 {
-	int x, y;
+	int x, y, nav_width;
+
+	if (NAV_WIDTH == 0)
+		return 1;
 
 	win_cursor_pos(&win, &x, &y);
-	return MAX(0, MIN(2, (x / (win.w * 0.33))));
+	nav_width = NAV_IS_REL ? win.w * NAV_WIDTH / 100 : NAV_WIDTH;
+	nav_width = MIN(nav_width, (win.w + 1) / 2);
+
+	return x < nav_width ? 0 : x < win.w - nav_width ? 1 : 2;
```

Comment:
Feel like this hurts readability a bit. But the existing `return MAX(0, MIN(2, (x / (win.w * 0.33))));` wasn't much better. I can let this slide, but I would prefer to break it up.

```suggestion
	if (x < nw)
		return 0;
	else
		return x < (win.w - nw) ? 1 : 2;
```

Reply:
I think it's okay as it is. Since the code logic is linear (it's the same order as how we are thinking). The readability is not that bad.

Reply:
Maybe even
```c
	if (x < nw)
		return 0;
	else if (x < win.w-nw)
		return 1;
	else
		return 2;
```

But that might be overkill.

Reply:
@eylles  what do you think?

Reply:
> I think it's okay as it is. Since the code logic is linear (it's the same order as how we are thinking). The readability is not that bad.

Let's wait for @eylles 's opinion on it.

- - -

But *if* we do want to keep it as is, then I would advise either using paren `x < (win.w - nav_width)` or grouping it together `x < win.w-nav_width`.

Reply:
I say we add parenthesis around the second `?:` `nav_width ? 0 : (x < win.w - nav_width ? 1 : 2)`

Reply:
> I say we add parenthesis around the second ?: x < nav_width ? 0 : (x < win.w - nav_width ? 1 : 2)

I could be okay with that. Would still advise grouping `win.w-nw`.

But anyways, let's wait for some more opinion. Not under any hurry.

Reply:
Hmmm you say the second suggestion may be overkill but to me it seems the clearest way to compare the widths.

Reply:
So we got 2 votes for the `if .. else if .. else` and 1 for chaining together `? :`